core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -1450,12 +1457,89 @@ class GroupCoordinator(val brokerId: Int,
             group.maybeInvokeJoinCallback(member, joinResult)
             completeAndScheduleNextHeartbeatExpiration(group, member)
             member.isNew = false
+
+            group.addPendingSyncMember(member.memberId)
           }
+
+          schedulePendingSync(group)
         }
       }
     }
   }
 
+  private def maybeRemovePendingSyncMember(
```

Comment:
nit: `removePendingSyncMember` throws an exception if the member is not in the group, so does the "maybe" in the name make sense?